enhancements/sig-architecture/6-placements/README.md

```diff
@@ -0,0 +1,470 @@
+	// If the slice is empty, clusters will be selected from the ManagedClusterSets bound to the placement
+	// namespace, otherwise clusters will be selected from the intersection of this slice and the
+	// ManagedClusterSets bound to the placement namespace.
+	// +optional
+	ClusterSets []string `json:"clusterSets,omitempty"`
+
+	// Name represents a selector of clusters by name.
+	// This field is mutually exclusive with other fields in ClusterPredicate.
+	// +optional
+	ClusterName string `json:"clusterName,omitempty"`
+
+	// NumberOfClusters represents the number of clusters to be selected which meet the
+	// requirements in this predicate.
+	// Nil indicates that NumberOfClusters is not set and there is no bound number.
+	// +optional
+	NumberOfClusters *int32 `json:"numberOfClusters,omitempty"`
+
+	// RequiredDuringSchedulingRequiredDuringExecution represents a scheduling strategy for a
+	// cluster selector. If specified, the cluster selector should be mached at both scheduling
+	// time (when the placement decisions are made for the first time) and execution time (After
+	// one or multiple placement decisions are made). It means,
+	// 1) Any cluster, which does not match the selector, should not be selected;
+	// 2) If a selected cluster ceases to match the selector (e.g. due to an update) after the
+	//    placement decisions are made, it will be eventually removed from the placement decisions;
+	// 3) If a cluster starts to match the selector after the placement decisions are made, it
+	//    will either be selected or at least has chance to be selected (when NumberOfClusters is
+	//    specified);
+	//
+	// Other scheduling strategies not yet implemented includes
+	// RequiredDuringSchedulingIgnoredDuringExecution and PreferredDuringSchedulingIgnoredDuringExecution
+	// +optional
+	RequiredDuringSchedulingRequiredDuringExecution *ClusterSelector `json:"requiredDuringSchedulingRequiredDuringExecution,omitempty"`
+}
+
+// ClusterSelector represents the AND of the containing selectors
+type ClusterSelector struct {
+	// LabelSelector represents a selector of clusters by label
+	// +optional
+	LabelSelector metav1.LabelSelector `json:"labelSelector,omitempty"`
+
+	// ClaimSelector represents a selector of cluster by clusterclaims
+	// +optional
+	ClaimSelector ClusterClaimSelector `json:"claimSelector,omitempty"`
+}
+
+// A clusterclaim selector is a claim query over a set of managed cluster. The result of matchClaims and
+// matchExpressions are ANDed. An empty cluster claim selector matches all objects. A null
+// cluster claim selector matches no objects.
+type ClusterClaimSelector struct {
+	// matchClaims is a map of {key,value} pairs. A single {key,value} in the matchClaims
+	// map is equivalent to an element of matchExpressions, whose key field is "key", the
+	// operator is "In", and the values array contains only "value". The requirements are ANDed.
+	// +optional
+	MatchClaims map[string]string `json:"matchClaims,omitempty"`
```

Comment:
You often see this in APIs when selecting nodes, but why have both options in `ClusterClaimSelector`?

Reply:
MatchClaims is removed.